ansible_events/drools/durable_rules_engine.py

```diff
@@ -0,0 +1,230 @@
+import json
+import logging
+
+import requests
+
+_logger = logging.getLogger(__name__)
+
+
+class DurableRulesEngine:
+    __host = None
+    __last_resp = None
+
+    def __init__(self, host):
+        self.__host = host
+
+    def create_ruleset(self, ruleset_name, ruleset_string):
+        req = {ruleset_name: json.loads(ruleset_string)}
+
+        r = requests.post(
+            self.__host + "/create-durable-rules-executor", json=req
+        )
+        if r.status_code != 200:
+            raise Exception(
+                f"Invalid status code: {r.status_code} - {r.reason}\n"
+                + json.loads(r.content)["details"]
+            )
+
+        id = r.text
+        return id
+
+    def assert_event(self, session_id, serialized_fact):
+        _logger.warning("assert_event not yet implemented: using assert_fact")
+        return self.assert_fact(session_id, serialized_fact)
+
+    def assert_fact(self, session_id, serialized_fact):
+        d = json.loads(serialized_fact)
+        if "j" in serialized_fact:
+            d["j"] = 1
+
+        serialized_fact = json.dumps(d)
+
+        r = requests.post(
+            f"{self.__host}/rules-durable-executors/{session_id}/process",
+            json=json.loads(serialized_fact),
+        )
+
+        if r.status_code != 200:
+            raise Exception(
+                f"Invalid status code: {r.status_code} - {r.reason}\n"
+                + json.loads(r.content)["details"]
+            )
+
+        self.__last_resp = r.json()
+        self.__last_resp.reverse()
+
+        return (0, session_id)
+
+    def start_action_for_state(self, handle):  # real signature unknown
+        try:
+            resp = self.__last_resp.pop()
+        except Exception:
```

Comment:
Too broad exception handler. Please specify the exact exception classes that are expected here.